### PR TITLE
contract hourly section

### PR DESCRIPTION
**Pull request by brendansudol**
adds an hourly section to the contractor resource section so that users can toggle between adding totals by year OR adding hours and rates (by year) and having the totals computed based on those numbers.

preview:
https://cl.ly/tHeg  

addresses and closes: https://github.com/18F/cms-hitech-apd/issues/862

### This pull request is ready to merge when...
- [x] Tests have been updated (and all tests are passing)
- [x] This code has been reviewed by someone other than the original author

### This feature is done when...
- [ ] Product has approved the experience

### CODE REVIEW

**Comment by jeromeleecms** [comment]
@mgwalker 
1) Yes. Let's capture everything if we can. In my reporting/analytics brain, it would nice to be able to trend hourly rates over time or across different areas.
2) Not that I can think of. If we are calculated the totals from the hours/rate they provide, it will still roll into the totals that you are using to calculate the other tables.

Hope that answers your questions!

File: web/src/containers/ActivityDetailContractorResources.js [hunk 1]
```diff
@@ -215,11 +240,80 @@ class ContractorExpenses extends Component {
                         label={year}
                         value={contractor.years[year]}
                         onChange={this.handleYearChange(i, year)}
+                        disabled={contractor.hourly.useHourly}
```
Comment (mgwalker) [inline web/src/containers/ActivityDetailContractorResources.js]: Kids today will never know how frickin' hard this used to be.

File: web/src/containers/ActivityDetailContractorResources.js [hunk 2]
```diff
@@ -215,11 +240,80 @@ class ContractorExpenses extends Component {
...
                     ))}
                   </div>
                 </div>
+                <div className="mb3 md-flex">
+                  <Label>
+                    {t('activities.contractorResources.labels.hourly')}
+                  </Label>
+                  <div className="md-col-6">
+                    <div className="flex items-center">
+                      <div className="mr2">This is an hourly resource</div>
```
Comment (mgwalker) [inline web/src/containers/ActivityDetailContractorResources.js]: Resource text.  I'm comfortable just making a note of this and fixing it later.  Can be part of our catch-up sprint.

File: web/src/reducers/activities.js [hunk 3]
```diff
@@ -281,6 +287,34 @@ const reducer = (state = initialState, action) => {
...
+      // if switching to hourly, clear out yearly totals
+      // if switch to yearly, clear out hourly numbers
+      const contractorResources = contractorsOld.map(
+        contractor =>
+          contractor.key !== contractorKey
+            ? contractor
+            : {
+                ...contractor,
+                ...(useHourly && {
+                  years: arrToObj(years, contractorDefaultYear())
```
Comment (mgwalker) [inline web/src/reducers/activities.js]: Clever

File: web/src/containers/ActivityDetailContractorResources.js (hunk 2 above)
Comment (brendansudol) [reply web/src/containers/ActivityDetailContractorResources.js]: 👍

File: web/src/containers/ActivityDetailContractorResources.js (hunk 1 above)
Comment (brendansudol) [reply web/src/containers/ActivityDetailContractorResources.js]: 😆

**Pull request merged**